idalloc/idalloc.go

```diff
@@ -40,6 +40,17 @@ func New() *IDAllocator {
 	}
 }
 
+func (a *IDAllocator) ReserveWellKnownID(id string, n uint64) {
+	if _, exists := a.uint64ToStr[n]; exists {
+		log.Panicf("ID %v already in use for well-known reservation '%v'", n, id)
```

Comment:
Is it right to panic inside the allocator?

Reply:
Seems OK to me.  This would be a start-of-day coding error, if it happened.